src/ValueObject/MicrosoftConfig.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace DoutorFinancas\MicrosoftGraphEmail\ValueObject;
+
+class MicrosoftConfig
+{
+    /**
+     * @var string
+     */
+    private $tenantId;
+    /**
+     * @var string
+     */
+    private $clientId;
+    /**
+     * @var string
+     */
+    private $secret;
+    /**
+     * @var string
+     */
+    private $driverId;
+    /**
+     * @var string
+     */
+    private $graphEndpoint;
+    /**
+     * @var string
+     */
+    private $authTokenURL;
+    /**
+     * @var string
+     */
+    private $scope;
+    /**
+     * @var string
+     */
+    private $grantType;
+    /**
+     * @var string
+     */
+    private $authenticationFlow;
+    /**
+     * @var string
+     */
+    private $graphApiVersion;
+    /**
+     * @var string|null
+     */
+    private $sharePointBaseUrl;
+    /**
+     * @var string|null
+     */
+    private $sharePointUserSpaceName;
+
+    public function __construct(
+        string $tenantId,
+        string $clientId,
+        string $secret,
+        string $driverId,
+        string $sharePointBaseUrl = null,
+        string $sharePointUserSpaceName = null,
+        string $graphEndpoint = 'https://graph.microsoft.com',
+        string $graphApiVersion = 'v1.0',
+        string $authTokenURL = 'https://login.microsoftonline.com/%s/oauth2/v2.0/token',
+        string $scope = 'https://graph.microsoft.com/.default',
+        string $grantType = 'client_credentials',
+        string $authenticationFlow = 'application' // just to know, not used in any place yet.
+    )
+    {
```

Comment:
Sounds good for me, but i think that I'll need to know more about these value types of Microsoft Graph, to avoid to write validations which could return false positives.